Content.Server/Medical/HealingSystem.cs

```diff
@@ -159,6 +166,25 @@ private bool TryHeal(EntityUid uid, EntityUid user, EntityUid target, HealingCom
         return true;
     }
 
+    /// <summary>
+    /// Scales the self-heal penalty based on the amount of damage taken
+    /// </summary>
+    /// <param name="uid"></param>
+    /// <param name="component"></param>
+    /// <returns></returns>
+    public float GetScaledHealingPenalty(EntityUid uid, HealingComponent component)
+    {
+        var output = component.Delay;
+        if (!TryComp<MobStateComponent>(uid, out var mobState) || !TryComp<DamageableComponent>(uid, out var damageable))
+            return output;
+
+        _mobStateSystem.TryGetEarliestCriticalState(mobState, 0, out var _, out var amount);
+        var percentDamage = (float) (damageable.TotalDamage / amount);
+        //basically make it scale from 1 to the multiplier.
+        var modifier = percentDamage * (component.SelfHealPenaltyMultiplier - 1) + 1;
```

Comment:
You should add a check somewhere before this to ensure that the multiplier or modifier is always positive, at least (even though yes, this is technically positive as long as the multiplier is greater than or equal to zero, and at the moment critical always incapacitates a user). Linear function for this is fine enough.